docs/troubleshooting.asciidoc

```diff
@@ -0,0 +1,85 @@
+== Troubleshooting
+
+[[no-data-sent]]
+=== No data is sent to the APM Server
+
+If neither errors or performance metrics are being sent to the APM Server,
+it's a good idea to first check your log file and look for output just as the app starts.
+
+[[error-messages]]
+==== Explanation of error messages
+
+[[message-agent-inactive]]
+===== Agent Inactive
+
+----
+Elastic APM agent is inactive due to configuration
+----
+
+You've most likely set the `active` option or the `ELASTIC_APM_ACTIVE` environment variable to `false`.
+Ensure that it's set to `true` and try again.
+
+See the API documentation about <<active,`active`>> for details.
+
+[[message-missing-app-name]]
+===== Missing `appName`
+
+----
+Elastic APM isn't correctly configured: Missing appName
+----
+
+Ensure that the agent is correctly configured with an `appName` config option either by passing it to the `start()` function,
+by specifying it in the optional configuration file,
+or by using environment variables.
+
+See the API documentation on <<app-name,`appName`>> for details.
+
+[[message-invalid-app-name]]
+===== Invalid `appName`
+
+----
+Elastic APM isn't correctly configured: appName "..." contains invalid characters! (allowed: a-z, A-Z, 0-9, _, -, <space>)
+----
+
+The `appName` you provided contained invalid characters.
+You can only use the the letters a-z (both upper and lowercase), numbers, underscore (`_`), hyphen (`-`), and space (` `).
+
+See the API documentation on <<app-name,`appName`>> for details.
+
+[[missing-performance-metrics]]
+=== No performance metrics sent to APM Server
+
+Errors gets tracked just fine,
+but you don't see any performance metrics.
+
+First make sure you have https://www.npmjs.com/package/elastic-apm[the latest version] of the Elastic APM Node.js agent.
+Issues are fixed and support for new metrics are added all the time,
+so make sure you keep your dependency up to date.
+
+If you are using the latest version but still don't see any performance metrics,
+make sure that the agent is *both required and started at the very top of your main app file* (usually the `index.js`, `server.js` or `app.js` file).
+It's important that the agent is started before any other modules are required.
+If not,
+the agent will not be able to hook into any modules and will not be able to measure the performance of your application.
+
+IMPORTANT: If you are using Babel / ES modules in your application,
+make sure you have read about the <<es-modules,Babel / ES Module support>>.
+
+If this is not the issue,
+your app dependencies might be incompatible with the agent.
+Please create a new topic in the https://discuss.elastic.co/c/apm[Elastic APM discuss forum] and try to include as much information as possible, e.g:
+
+* The dependencies in you `package.json` file
+* Your apps main js file showing how the agent is initialized (usually the `index.js`, `server.js` or `app.js` file)
+* Your server logs with Elastic APM debugging enabled (see below for details)
```

Comment:
Stuff tends to move around over time, so avoid saying things like "see below for details." Instead, just add a link. It doesn't matter if the content immediately follows the link.